src/utils/fsUtils.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import { Position, TextDocument, TextEditor, TextEditorEdit, Uri, ViewColumn, window, workspace } from "vscode";
+import { localize } from './localize';
+
+export namespace fsUtils {
+    export async function getUniqueFileName(folderPath: string, prefix: string, suffix?: string): Promise<string> {
+        let count: number = 0;
+        const maxCount: number = 1024;
+
+        while (count < maxCount) {
+            const fullFileName: string = `${prefix}${count === 0 ? '' : `-${count.toString()}`}${suffix}`;
+
+            const fullPath: string = path.join(folderPath, fullFileName);
+
+            const pathExists: boolean = await fse.pathExists(fullPath);
+            const editorExists: boolean = workspace.textDocuments.some((doc: TextDocument) => isPathEqual(doc.uri.fsPath, fullPath));
+            if (!pathExists && !editorExists) {
+                return fullFileName;
+            }
+
+            count += 1;
+        }
+
+        throw new Error(localize('failedUnique', 'Failed to find unique name for new file.'));
+    }
+
+    export function isPathEqual(fsPath1: string, fsPath2: string): boolean {
+        return path.relative(fsPath1, fsPath2) === '';
+    }
+
+    export async function showNewFile(data: string, fileNamePrefix: string, fileNameSuffix: string, viewColumn?: number): Promise<void> {
```

Comment:
I recently improved this code a bit...

export async function showNewFile(data: string, extensionPath: string, fileName: string, fileExtension: string, column?: vscode.ViewColumn): Promise<void> {
    let uri: vscode.Uri;
    const folderPath: string = vscode.workspace.rootPath || extensionPath;
    const fullFileName: string | undefined = await getUniqueFileName(folderPath, fileName, fileExtension);
    uri = vscode.Uri.file(path.join(folderPath, fullFileName)).with({ scheme: 'untitled' });
    const textDocument = await vscode.workspace.openTextDocument(uri);
    const editor = await vscode.window.showTextDocument(textDocument, column ? column > vscode.ViewColumn.Three ? vscode.ViewColumn.One : column : undefined, true);
    await writeToEditor(editor, data);
}

Reply:
Uhh I can't tell what's different and I didn't just copy this code straight from Cosmos so I can't just paste what you have. What did you improve?